### PR TITLE
physics integration and server sided clean up

### PR DESCRIPTION
**Pull request by KNOXDEV**
Things this PR does:

- [x] I noticed the physics engine shares some logic duplication with the new checkpoint system. I've abstracted out bounding box and collision detection into its own module and its now shared.
- [x] added comments and abstracted out functions in the main server loop
- [x] Simplified the GamePhase enum a little bit which helps some verbosity.

Everything is essentially the same just shuffled around a bit. Note its all server side atm.

### CODE REVIEW

File: chariot-server/src/game/phase.rs [hunk 1]
```diff
@@ -29,46 +29,25 @@ use chariot_core::networking::Uuid;
  *  Show standings, perhaps a retry button, any other end-of-race stuff
  */
 
-#[derive(Debug)]
 pub enum GamePhase {
```
Comment (TheCrossBoy) [inline chariot-server/src/game/phase.rs:32]: I like this change a lot very cool

**Review by trulyronak — COMMENTED** [review]
Game phase logic (what I wrote) looks good to me cool stuff I learned more rust reading this

**Review by TheCrossBoy — APPROVED** [review]
aaron james eason strikes yet again with a banger pr

**Review by alistairgray42 — APPROVED** [review]
Physics sounds good, assuming that you didn't sneakily add some stray minus signs or something

**Comment by KNOXDEV** [comment]
> Physics sounds good, assuming that you didn't sneakily add some stray minus signs or something

tests still pass!

**Pull request merged**